src/WordPress/WordPressAPI.php

```diff
@@ -69,6 +70,48 @@ public function getHostAPIKey()
     }
 
     /**
+     * We wrap the return value with an array to be consistent between
+     * other plugins.
+     * 
+     * @param null $userId
+     *
+     * @return mixed
+     */
+    public function getDomainList($userId = null)
+    {
+        $cachedDomainName = $this->dataStore->getDomainNameCache();
+        if (empty($cachedDomainName)) {
```

Comment:
The logic is [here](https://github.com/cloudflare/CloudFlare-Wordpress/blob/f0bda21af307943e7da0f727eceed7fdea5d4e38/src/WordPress/WordPressAPI.php#L155-L168). 

1. If the cached value and the original domain are not equal
2. Cache it

so even if `$cachedDomainName` in `getDomainList` returns empty the cache will be set to `getOriginalDomain()`. I wanted `DataStore` to only fetch and pass data so I put the logic in `WordPressAPI`.

The reason I added `if (empty($cachedDomainName))` is I wanted to make [this](https://github.com/cloudflare/CloudFlare-Wordpress/blob/f0bda21af307943e7da0f727eceed7fdea5d4e38/src/WordPress/WordPressAPI.php#L159) check.